identity-iota/src/credential/validation_options.rs

```diff
@@ -0,0 +1,130 @@
+  #[serde(default)]
+  pub latest_issuance_date: Option<Timestamp>,
+  /// Declare that the credential's signature must be verified according to these
+  /// [VerifierOptions].
+  #[serde(default)]
+  pub verifier_options: VerifierOptions,
+}
+
+impl CredentialValidationOptions {
+  /// Constructor that sets all options to their defaults.
+  pub fn new() -> Self {
+    Self::default()
+  }
+
+  /// Declare that a credential may expire no later than the given `timestamp`.
+  pub fn earliest_expiry_date(mut self, timestamp: Timestamp) -> Self {
+    self.earliest_expiry_date = Some(timestamp);
+    self
+  }
+  /// Declare that a credential may expire no later than the given `timestamp`.
+  pub fn latest_issuance_date(mut self, timestamp: Timestamp) -> Self {
+    self.latest_issuance_date = Some(timestamp);
+    self
+  }
+
+  /// Declare that the signature of a credential is to be verified according to the given
+  /// `options`.
+  pub fn verifier_options(mut self, options: VerifierOptions) -> Self {
+    self.verifier_options = options;
+    self
+  }
+}
+
+/// Declares how a credential subject must relate to the presentation holder.
+///
+/// See [`PresentationValidationOptions::subject_holder_relationship`](PresentationValidationOptions::
+/// subject_holder_relationship()).
+// Need to use serde_repr to make this work with duck typed interfaces in the Wasm bindings.
+#[derive(Debug, Clone, Copy, serde_repr::Serialize_repr, serde_repr::Deserialize_repr)]
+#[repr(u8)]
+pub enum SubjectHolderRelationship {
+  /// Declare that the holder must always match the subject.
+  AlwaysSubject = 0,
+  /// Declare that the holder must match the subject on credentials with the nonTransferable property set.
+  SubjectOnNonTransferable = 1,
+  /// Declares that the subject is not required to have any kind of relationship to the holder.  
+  Any = 2,
+}
+
+impl Default for SubjectHolderRelationship {
+  fn default() -> Self {
+    // TODO: Should the default be the most strict variant?
+    Self::SubjectOnNonTransferable
+  }
+}
+
+/// Declares when validation should return with an error.
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+pub enum FailFast {
+  /// Declare that validation should fail after all errors have been found
+  No,
+  /// Declare that validation must fail upon the first error is found
+  Yes,
+}
+
+/// Options to declare validation criteria for validation methods such as
+/// [`PresentationValidator::full_validation`](super::PresentationValidator::full_validation()) and
+/// [`Resolver::verify_presentation`](crate::tangle::Resolver::verify_presentation()).
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct PresentationValidationOptions {
+  /// Declares that the credentials of the presentation must all be
+  /// validated according to these [`CredentialValidationOptions`].
+  #[serde(default)]
+  pub shared_validation_options: CredentialValidationOptions,
```

Comment:
I have changed the documentation as suggested, but I am not quite sure about changing the name however. I feel the fact that it is for credentials is reflected in the type and `shared` quickly conveys the fact that it is for all credentials. I have also considered the possibility of extending this struct in the future to also allow for credential validation option specialization. Not exactly sure about the representation/API of that at this point, but it could be something like a map with keys issuer DID's and values CredentialValidationOptions. If we want to support something like that then we would maybe have to rethink the names anyway (or accept slightly sub-optimal names to avoid a breaking change).

Reply:
I'm fine leaving the name as-is. I definitely agree about the "shared" part, just found it slightly difficult to see at-a-glance what this field was in the examples. I also considered `shared_credential_options`, not sure if that's any better.